### ISSUE TITLE
RPI: Not able to complete Installation

### ISSUE THREAD

**Issue by itchy2** [timothycrosley/streamdeck-ui#37]
Followed the installation manual:

```
pi@raspberrypi:~ $ pip3 install streamdeck_ui
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting streamdeck_ui
  Using cached https://files.pythonhosted.org/packages/98/4d/fe60feaa7b0d5c7afac6da68cfb8b4fd5f48a10e3f9824e3764e7301a5a4/streamdeck_ui-1.0.2-py3-none-any.whl
Collecting pynput<2.0,>=1.4 (from streamdeck_ui)
  Using cached https://files.pythonhosted.org/packages/33/0a/ea13c055a90b1aff5945e7eb330584f15e5282aead15a8f3cdb977a1534e/pynput-1.6.8-py2.py3-none-any.whl
Collecting pyside2<6.0,>=5.13 (from streamdeck_ui)
  Could not find a version that satisfies the requirement pyside2<6.0,>=5.13 (from streamdeck_ui) (from versions: )
No matching distribution found for pyside2<6.0,>=5.13 (from streamdeck_ui)

```

**Comment by sftim** [comment]
@itchy2, what operating system were you running on your Raspberry Pi? That detail is going to be important to help understand what is wrong.

**Comment by yenoromm** [comment]
For those that come across this, it is the missing support for PySide2 on ARM. There are a few workarounds listed including running Arch on the RPI or install parts of PySide2 via Aptitude packages however I opted for `usbip` on the RPI and a Linux instance on my server as it was less work for me.

**Comment by dodgyrabbit** [comment]
Thank you for reporting this - I think what would be most useful is if someone wants to take a stab at writing up a markdown file on how to get this to run on an RPI. There were a few issues like this opened. Closing this for now.

**Comment by new2f7** [comment]
@yenoromm There is PySide2 support for Raspberry Pi OS: https://forum.qt.io/topic/112813/installing-pyside2-on-raspberry-pi
I've installed it via `sudo apt install python3-pyside2.*`

However, I'm getting the following error messages:
![streamdeck-ui_raspberry-pi](https://user-images.githubusercontent.com/14283108/155705027-ec6c93a8-82bc-4d89-8d40-761aa4094611.jpg)

@dodgyrabbit I would love to see streamdeck-ui support for Raspberry PI OS and this issue reopened (and wontfix removed). If you need any help or additional information let me know. I would like to use it for a project in three days and are willing to help.

**Comment by yenoromm** [comment]
> @yenoromm There is PySide2 support for Raspberry Pi OS: https://forum.qt.io/topic/112813/installing-pyside2-on-raspberry-pi I installed it via `sudo apt install python3-pyside2.*`


If you read my comment again, you’ll see I did mention that some parts of PySide2 can be installed with APT. I too got the same error back when I tried that method and couldn’t get around it.

**Comment by dodgyrabbit** [comment]
@new2f7 Can you verify which version of streamdeck-ui you have installed? The call stack suggests it's not the latest version (1.1.3). I'm not saying it will fix it, but start there.

WRT troubleshooting. The error `KeyError: None` basically means it could not find the stream deck serial number in the deck[] dictionary. The initialization fo that dictionary is done earlier in the code but not knowing which version you have installed makes it difficult to make progress.

**Comment by new2f7** [comment]
The version via pip that I used above is very old (0.0.2).

I installed the master version via `pip install git+https://github.com/new2f7/streamdeck-ui` (PySide2 dependency [disabled](https://github.com/new2f7/streamdeck-ui/commit/bf25b94ae036c1f86a371d914ff4f427e9ce0353)).
Now the UI launches and it seems to work. There is only one warning when launching:
`libEGL warning: DRI2: failed to authenticate`
